utils/check_content_style.py

```diff
@@ -0,0 +1,173 @@
+# this program. If not, see <https://www.gnu.org/licenses/>.
+
+import re
+import glob
+
+# Script that checks for common text content formatting pitfalls not covered by spellcheck or other tests.
+
+# The number of errors found in all files
+errors = 0
+# The list of errors in the current file; elements are tuples of the line numbers and the error messages
+error_list = []
+# Regex-based style checks for check_regexes()
+regexes = {
+	# Matches any space that is preceeded only by tabs or " or ` characters,
+	# except if it immediately follows a " or ` character.
+	"^[\\t`\"]*(?<![`\"]) ": "indentations should use tabs only",
+	# Matches multiple space characters.
+	"  ": "multiple consecutive space characters",
+	"\\s$": "trailing whitespace character"
+}
+# Precompile regexes
+regexes = {re.compile(regex): description for regex, description in regexes.items()}
+
+
+# Checks the format of all txt files.
+def check_content_style():
+	files = glob.glob('data/**/*.txt', recursive=True)
+	files.sort()
+	for file in files:
+
+		f = open(file, "r")
+		lines = f.readlines()
+		lines = [line.removesuffix('\n') for line in lines]
+
+		check_copyright(file, lines)
+		check_ascii(file, lines)
+		check_regexes(file, lines)
+		check_indents(file, lines)
+
+		# Making sure errors are printed in order of their line numbers
+		error_list.sort(key=lambda error: error[0])
+		for (line, error) in error_list:
+			print(error)
+		error_list.clear()
+
+
+# Checks the copyright header of the files. Unlike C++ source files, these don't contain the file's name in the header.
+# The tooltips.txt file is excluded from this check, as it has no copyright header.
+# Parameters:
+# file: the name of the file
+# lines: the lines of the file
+def check_copyright(file, lines):
+	if file == "data/tooltips.txt":
+		return
+	# The mandatory first line of a copyright header; might be repeated
+	first = "^# Copyright \\(c\\) \\d{4}(?:(?:-|, )\\d{4})? by .+$"
+	# Optional line following the first one; might be repeated
+	second = "^# Based on earlier text copyright \\(c\\) \\d{4}(?:(?:-|, )\\d{4})? by .+$"
+	# The final segment of the copyright header
+	third = """\
+	#
+	# Endless Sky is free software: you can redistribute it and/or modify it under the
+	# terms of the GNU General Public License as published by the Free Software
+	# Foundation, either version 3 of the License, or (at your option) any later version.
+	#
+	# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+	# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+	# PARTICULAR PURPOSE. See the GNU General Public License for more details.
+	#
+	# You should have received a copy of the GNU General Public License along with
+	# this program. If not, see <https://www.gnu.org/licenses/>.
+	\
+	""".splitlines()
+	text_index = 0
+	for line in lines:
+		if re.search(first, line):
+			text_index += 1
+		else:
+			break
+	if text_index == 0:
```

Comment:
Yeah, this is one of the worst-looking parts of the code. The thing is, the `first` regex may match multiple times, since copyright headers can contain more than one copyright holders, so I can't assume it is only going to be the first item in the list.